### PR TITLE
docs: add examples and docs for facet

### PR DESCRIPTION
**Pull request by Zehua-Zeng**

### CODE REVIEW

File: docs/facts/examples.ipynb [hunk 1]
```diff
@@ -693,6 +693,167 @@
...
+    "    },\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6c136876",
+   "metadata": {},
+   "source": [
+    "### Faceted Scatterplot on Horizontal Direction with Binning"
```
Comment (domoritz) [inline docs/facts/examples.ipynb:786]:
```suggestion
    "### Facet Scatterplot by Binned Data into Columns"
```

File: docs/facts/examples.ipynb [hunk 2]
```diff
@@ -693,6 +693,167 @@
...
+    "\n",
+    "The facet option produces horizontal or vertical small multiples to compare grouped data."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "94641063",
+   "metadata": {},
+   "source": [
+    "### Faceted Scatterplot on Horizontal Direction"
```
Comment (domoritz) [inline docs/facts/examples.ipynb:711]:
```suggestion
    "### Facet Scatterplot into Columns"
```

**Review by domoritz — CHANGES_REQUESTED** [review]
Make sure to run `make` before committing.

File: docs/facts/examples.ipynb [hunk 3]
```diff
@@ -713,7 +874,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 46,
+   "execution_count": null,
```
[the remark this answers is not in the document]
Comment (Zehua-Zeng) [reply docs/facts/examples.ipynb]: Seems like running make doesnt help, need to run all examples manually

File: docs/facts/examples.ipynb [hunk 4]
```diff
@@ -340,39 +340,24 @@
...
-      ]
-     },
-     "execution_count": 37,
-     "metadata": {},
-     "output_type": "execute_result"
+     "ename": "AssertionError",
+     "evalue": "Invalid spec",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
```
Comment (domoritz) [inline docs/facts/examples.ipynb:355]: Fix this error

**Review by domoritz — CHANGES_REQUESTED** [review]
Please also add documentation and something in generate before we merge this.

**Review by domoritz — APPROVED** [review]
Looks good. In a follow up pull request add constraints to restrict binning (and require it) for numeric fields.

**Pull request merged**